### PR TITLE
MCLOUD-3059: [Spike] Investigate issue with elasticsearch image on AWS

### PR DESCRIPTION
**Pull request by arhiopterecs**
### Description
Added option `--es-env-var` for `docker-compose.yml` file generator. Using this option, you can specify the environment variables that will be applied when starting the Elasticsearch container.
This will allow you to change service settings, such as heap size for JVM or Elasticsearch options.
Parameter changes for JVM are available from image version 1.7
Changing Elasticsearch parameters using environment variables is available starting from version 5.2

### Fixed Issues (if relevant)
1. https://jira.corp.magento.com/browse/MCLOUD-3059

### Manual testing scenarios
1. Clone `[magento-cloud template](https://github.com/magento/magento-cloud)`.
2. Run: `$ composer update` in the magento-cloud template directory: 
3.  Generate the docker-compose.yml file with env variables for `elasticsearch` service: 
```bash
$ php vendor/bin/ece-docker build:compose --es-env-var=ES_JAVA_OPTS="-Xms512m -Xmx512m" --es-env-var=node.store.allow_mmapfs=false
Configuration was built.
```
NOTE: By default, an image with Elasticsearch version 6.5 is used. For this version of Elasticsearch, the option `node.store.allow_mmapfs` exists.
For image varsions 5.2, 2.4, 1.7 ,this option does not exist
Forimage varsions 7.5, this parameter was renamed to node.store.allow_mmap
4. Check the file `docker-compose.yml`, the following settings must exist:
```
... 
elasticsearch:
    hostname: elasticsearch.magento2.docker
    image: 'magento/magento-cloud-docker-elasticsearch:6.5-1.1'
    environment:
      - 'ES_JAVA_OPTS=-Xms512m -Xmx512m'
      - node.store.allow_mmapfs=false
    networks:
      magento:
        aliases:
          - elasticsearch.magento2.docker           
....  
```
5. Run: `$ docker-compose up -d`. Containers started successfully
6. Login in elasticsearch container:
```
$ docker-compose exec elasticsearch bash
```
7. Run  the command in the container: `$ ps aux  | grep -E "(Xms|Xmx)"`
Expected: The last values ​​of the `-Xms` and` -Xmx` parameters are 512m

8. Run  the command in the container: ` $ curl http://localhost:9200/_nodes 2>&1 | grep 'mmapfs'`.
Expected:`"allow_mmapfs":"false"` should be present

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages

releasenotes
Added the `docker-compose --es-env-var` option to customize the Elasticsearch container configuration using environment variables, for example setting the heap size for JVM or other Elasticsearch options. See [Elasticsearch container](https://devdocs.magento.com/cloud/docker/docker-containers-service.html#elasticsearch-container).

### CODE REVIEW

**Comment by arhiopterecs** [comment]
PR was recreated https://github.com/magento/magento-cloud-docker/pull/174

**Review by shiftedreality — APPROVED** [review]

File: images/elasticsearch/2.4/Dockerfile [hunk 1]
```diff
@@ -1,11 +1,14 @@
 FROM elasticsearch:2.4
 
+RUN apt update -y && \
+    apt install procps -y
```
Comment (oshmyheliuk) [inline images/elasticsearch/2.4/Dockerfile:4]: Why did you add `procps` installation only for ES 2.4?  And where it's using?

Comment (arhiopterecs) [reply images/elasticsearch/2.4/Dockerfile:4]: @oshmyheliuk 
This image does not have the `ps` tool.
Needed for functional tests
https://github.com/magento/magento-cloud-docker/pull/174/files#diff-1fe73be4a3ae3c969d689c20c9934212R45

Comment (oshmyheliuk) [reply images/elasticsearch/2.4/Dockerfile:4]: Ok, thanks, for me was strange that only one image doesn't have `ps` tool.

**Review by oshmyheliuk — APPROVED** [review]

**Review by BaDos — APPROVED** [review]

**Comment by shiftedreality** [comment]
ES images must be re-built to fix the tests. Approved.
